runtime/lua/vim/highlight.lua

```diff
@@ -28,16 +28,30 @@ end
 --- use from init.vim via
 ---   au TextYankPost * lua require'vim.highlight'.on_yank()
 --- customize highlight group and timeout via
----   au TextYankPost * lua require'vim.highlight'.on_yank("IncSearch", 500)
+---   au TextYankPost * lua require'vim.highlight'.on_yank({higroup="IncSearch", timeout=250})
+--- customize conditions (do not highlight a visual selection) via
+---   au TextYankPost * lua require'vim.highlight'.on_yank({visual=false})
```

Comment:
Maybe what would be better is we expose `highlight` on `vim.highlight` directly, so we don't need to have the `require` business at all

Reply:
How would we do that?

Reply:
I don't remember off-hand, but you can see how `lsp` did it somewhere... I can look around later if you want. I'm on my laptop w/ only chrome setup, not for nvim dev, so it's hard for me to look it up. Or for other `vim.*` style things that are auto loaded. For example, `vim.inspect` is a good one. I think it's either `shared.lua` or maybe `vim.lua` somewhere.